storops/unity/calculator.py

```diff
@@ -295,6 +313,254 @@ def _sp_pct(path, curr, prev):
+                                           (delta_other1 + delta_other2))
+
+    else:
+        ret = IdValues()
+
+    return ret
+
+
+@metric_calculator
+def disk_queue_length(path, prev, curr):
+    if len(path) != 3:
+        raise ValueError(
+            'takes in "sumArrivalQueueLength", "reads" and '
+            '"writes" counter.')
+
+    part1, part2, part3 = path
+    if all_not_none(prev, curr):
+        prev_part1 = prev.by_path(part1)
+        prev_part2 = prev.by_path(part2)
+        prev_part3 = prev.by_path(part3)
+
+        curr_part1 = curr.by_path(part1)
+        curr_part2 = curr.by_path(part2)
+        curr_part3 = curr.by_path(part3)
+
+        delta_part1 = curr_part1.sum_sp() - prev_part1.sum_sp()
+        delta_part2 = curr_part2.sum_sp() - prev_part2.sum_sp()
+        delta_part3 = curr_part3.sum_sp() - prev_part3.sum_sp()
+
+        ret = delta_part1 / (delta_part2 + delta_part3)
+
+    else:
+        ret = IdValues()
+
+    return ret
+
+
+@metric_calculator
+def lun_response_time(path, prev, curr):
+    if len(path) != 3:
+        raise ValueError(
+            'takes in "totalIoTime", "reads" and "writes" counter.')
+
+    part1, part2, part3 = path
+    if all_not_none(prev, curr):
+        prev_part1 = prev.by_path(part1)
+        prev_part2 = prev.by_path(part2)
+        prev_part3 = prev.by_path(part3)
+
+        curr_part1 = curr.by_path(part1)
+        curr_part2 = curr.by_path(part2)
+        curr_part3 = curr.by_path(part3)
+
+        delta_part1 = curr_part1.sum_sp() - prev_part1.sum_sp()
+        delta_part2 = curr_part2.sum_sp() - prev_part2.sum_sp()
+        delta_part3 = curr_part3.sum_sp() - prev_part3.sum_sp()
+
+        ret = delta_part1 / (delta_part2 + delta_part3)
+
+    else:
+        ret = IdValues()
+
+    return ret
+
+
+@metric_calculator
+def lun_queue_length(path, prev, curr):
+    if len(path) != 3:
+        raise ValueError(
+            'takes in "currentIOCount", "busyTime" and '
+            '"idleTime" counter.')
+
+    part1, part2, part3 = path
+    if all_not_none(prev, curr):
+        prev_part1 = prev.by_path(part1)
+        prev_part2 = prev.by_path(part2)
+        prev_part3 = prev.by_path(part3)
+
+        curr_part1 = curr.by_path(part1)
+        curr_part2 = curr.by_path(part2)
+        curr_part3 = curr.by_path(part3)
+
+        delta_part1 = curr_part1.sum_sp() - prev_part1.sum_sp()
+        delta_part2 = curr_part2.sum_sp() - prev_part2.sum_sp()
+        delta_part3 = curr_part3.sum_sp() - prev_part3.sum_sp()
+
+        ret = delta_part1 * delta_part2 / (delta_part2 + delta_part3)
+
+    else:
+        ret = IdValues()
+
+    return ret
+
+
+@metric_calculator
+def sum_sp(path, _, curr):
+    path = only_one_path(path)
+    return curr.by_path(path).sum_sp()
+
+
+@metric_calculator
+def byte_rate(path, prev, curr):
+    if len(path) != 2:
+        raise ValueError('takes in "Blocks" and "blockSize" counter.')
+
+    part1, part2 = path
```

Comment:
Better to name them `blocks` and `blockSize`.  Otherwise the algorithm below is hard to understand.

Reply:
Done, already rename the variables.